src/msgtree.rs

```diff
@@ -0,0 +1,178 @@
+//   git-dit - the distributed issue tracker for git
+//   Copyright (C) 2016 Matthias Beyer <mail@beyermatthias.de>
+//   Copyright (C) 2016 Julian Ganz <neither@nut.email>
+//
+//   This program is free software; you can redistribute it and/or modify
+//   it under the terms of the GNU General Public License version 2 as
+//   published by the Free Software Foundation.
+//
+
+use git2::{Commit, Oid};
+use std::fmt::{self, Write};
+use std::iter::FromIterator;
+
+
+/// Representation of graph elements used to display trees
+///
+#[derive(Clone, PartialEq)]
+pub enum TreeGraphElem {
+    Empty,
+    Following,
+    Mark(MarkType),
+}
+
+/// Representation of the type of mark
+///
+#[derive(Clone, PartialEq)]
+pub enum MarkType {
+    Start,
+    Mid,
+    End
+}
+
+
+/// Representation of one line of tree graph elements
+///
+pub struct TreeGraphElemLine(Vec<TreeGraphElem>);
+
+impl TreeGraphElemLine {
+    /// Append a graph element to the line
+    pub fn append(&mut self, e: TreeGraphElem) {
+        self.0.push(e);
+    }
+
+    /// Reverse "start" and "end" marks
+    ///
+    pub fn reverse_marks(&mut self) {
+        for elem in self.0.iter_mut() {
+            match elem {
+                &mut TreeGraphElem::Mark(MarkType::Start) => *elem = TreeGraphElem::Mark(MarkType::End),
+                &mut TreeGraphElem::Mark(MarkType::End) => *elem = TreeGraphElem::Mark(MarkType::Start),
```

Comment:
Why not implementing a `fn reverse(self) -> MarkType` on `MarkType` for less complex-to-read code?